src/huggingface_hub/utils/_errors.py

```diff
@@ -46,48 +51,60 @@ class EntryNotFoundError(HTTPError):
-                f" {request_id})"
-            )
-        elif error_code == "RevisionNotFound":
-            raise RevisionNotFoundError(
-                f"404 Client Error: Revision Not Found for url: {request.url}. (Request"
-                f" ID: {request_id})"
-            )
-        elif error_code == "EntryNotFound":
-            raise EntryNotFoundError(
-                f"404 Client Error: Entry Not Found for url: {request.url}. (Request"
-                f" ID: {request_id})"
+    request_id = response.headers.get("X-Request-Id")
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if "X-Error-Code" in response.headers:
+            error_code = response.headers["X-Error-Code"]
+            if error_code == "RepoNotFound":
+                message = (
+                    f"{response.status_code} Client Error: Repository Not Found for"
+                    f" url: {response.url}. If the repo is private, make sure you are"
+                    f" authenticated. (Request ID: {request_id})"
+                )
+                raise RepositoryNotFoundError(message, response)
+            elif error_code == "RevisionNotFound":
+                message = (
+                    f"{response.status_code} Client Error: Revision Not Found for url:"
+                    f" {response.url}. (Request ID: {request_id})"
+                )
+                raise RevisionNotFoundError(message, response)
+            elif error_code == "EntryNotFound":
+                message = (
+                    f"{response.status_code} Client Error: Entry Not Found for url:"
+                    f" {response.url}. (Request ID: {request_id})"
+                )
+                raise EntryNotFoundError(message, response)
+
+        if response.status_code == 401:
+            # The repo was not found and the user is not Authenticated
+            message = (
+                f"{response.status_code} Client Error: Repository Not Found for url:"
+                f" {response.url}. If the repo is private, make sure you are"
+                f" authenticated. (Request ID: {request_id})"
             )
+            raise RepositoryNotFoundError(message, response)
 
-    if request.status_code == 401:
-        # The repo was not found and the user is not Authenticated
-        raise RepositoryNotFoundError(
-            f"401 Client Error: Repository Not Found for url: {request.url}. If the"
-            " repo is private, make sure you are authenticated. (Request ID:"
-            f" {request_id})"
-        )
+        if request_id is not None and len(e.args) > 0 and isinstance(e.args[0], str):
+            e.args = (e.args[0] + f" (Request ID: {request_id})",) + e.args[1:]
 
-    _raise_with_request_id(request)
+        raise e
 
 
-def _raise_with_request_id(request):
-    request_id = request.headers.get("X-Request-Id")
+def _raise_with_request_id(response):
```

Comment:
why do we have two distinct functions ` _raise_for_status` and `_raise_with_request_id`?

Reply:
would be better to always report the request-id if it's present no?

Reply:
`_raise_for_status` also reports the request id if present. I think the naming of the functions might be a bit misleading. Technically we could remove `_raise_with_request_id` as the fallback logic within `_raise_for_status` is the same. This is only used in `whoami`.